lib/aruco_detector.hpp

```diff
@@ -5,13 +5,24 @@
 #include <memory>
 #include "classification_model.hpp"
 
+struct ArucoParams {
+    double markerSize;
+    cv::Mat cameraMatrix;
+    cv::Mat distCoeffs;
+    std::vector<int> ids;
+    std::vector<std::vector<cv::Point2f>> corners;
+    std::vector<cv::Vec3d> rvecs;
+    std::vector<cv::Vec3d> tvecs;
+};
+
 class Aruco_Detector {
 public:
     Aruco_Detector(const cv::Mat& cameraMatrix, int dictionaryType, float markerSize = 0.15)
         : cameraMatrix(cameraMatrix), dictionaryType(dictionaryType), markerSize(markerSize) {}
 
     friend ClassificationModel::Detection model_classify(Aruco_Detector &detector, cv::Mat& image, float threshold) {
-        if (detector.detectMarkers(image)) {
+        ArucoParams params;
+        if (detector.detectMarkers(image, params)) {
```

Comment:
The `params` should be a class variable taken by constructor. It can be accessed by both friend and private variables, so there isn't a need to pass it as argument.